src/gazebo_mavlink_interface.cpp

```diff
@@ -417,30 +432,77 @@ void GazeboMavlinkInterface::handle_actuator_controls(const gz::sim::UpdateInfo
 
   last_actuator_time_ = _info.simTime;
 
-  for (unsigned i = 0; i < n_out_max; i++) {
-    input_index_[i] = i;
-  }
-  // Read Input References
-  input_reference_.resize(n_out_max);
-
   Eigen::VectorXd actuator_controls = mavlink_interface_->GetActuatorControls();
   if (actuator_controls.size() < n_out_max) return; //TODO: Handle this properly
 
-  for (int i = 0; i < input_reference_.size(); i++) {
+  // Read Cmd vel input for rover
+  if (actuator_controls[n_out_max - 1] != 0.0 || actuator_controls[n_out_max - 2] != 0.0) {
```

Comment:
Is this really safe enough to detect that the device is a rover?
I think it would be good to somehow detect the model other way. Is there any possibility to detect e.g. the plugin type connected to the actuator? Like gz::sim::systems::DiffDrive, gz::sim::systems::MulticopterMotorModel or gz::sim::systems::JointPositionController

Reply:
Found one way to read the motor info from sdf. Could it be solved with something like this?
https://github.com/tiiuae/px4-gzsim-plugins/commit/f3ce874dd07b7946b56cfbe27e453e2e29567557

Reply:
Example log prints:

Reply:
Thanks. That's a very good solution. I chagned it!